localstripe/resources.py

```diff
@@ -3277,3 +3277,66 @@ def __init__(self, card=None, customer=None, **kwargs):
 
         self.type = 'card'
         self.card = card_obj
+
+
+class UsageRecord(StripeObject):
+    object = 'usage_record'
+    _id_prefix = 'mbur_'
+
+    class _UsageRecordSummaryWrapper(dict):
+        def _export(self, expand=None):
+            return self
+
+    def __init__(self, subscription_item_id=None, quantity=None,
+                 timestamp=None):
+        self.quantity = quantity
+        self.subscription_item = subscription_item_id
+        self.timestamp = timestamp
+
+        super().__init__()
+
+    @classmethod
+    def _api_list_summaries(cls, subscription_item_id=None, limit=None,
+                            starting_after=None, ending_before=None):
+        subscription_item = SubscriptionItem._api_retrieve(
+            subscription_item_id
+        )
+
+        usage_records = [value for key, value in store.items()
+                         if key.startswith(cls.object + ':')
+                         and value.subscription_item == subscription_item_id]
```

Comment:
When I see this operation, and the routes below (`/v1/subscription_items/{subscription_item_id}/usage_records`, `/v1/subscription_items/{subscription_item_id}/usage_record_summaries`) I think it would be simpler and more logical to have usage record stored direcly inside the corresponding subscription item.


Also I'm OK with a new object `class UsageRecord()` but I think I'm NOK with a `class UsageRecord(StripeObject)`, because usage record are never used with the standard APIs that all other `StripeObject` implement, ie `api_create`, `api_retrieve`, `api_update`, `api_delete`, `api_list_all` and associated HTTP route and method (`GET  object/id`, `POST object/id` etc).

So I think that routes `/usage_records` and `/usage_record_summaries` should be extensions of the object subscription item.

What do you think?